src/txn/txn_recover.c

```diff
@@ -121,6 +121,20 @@ __txn_op_apply(
 	end = *pp + opsize;
 
 	switch (optype) {
+	case WT_LOGOP_COL_MODIFY:
+		WT_ERR(__wt_logop_col_modify_unpack(session, pp, end,
+		    &fileid, &recno, &value));
+		GET_RECOVERY_CURSOR(session, r, lsnp, fileid, &cursor);
+		cursor->set_key(cursor, recno);
+		if ((ret = cursor->search(cursor)) != 0)
+			WT_ERR_NOTFOUND_OK(ret);
+		else {
+			WT_ERR(__wt_modify_apply(
```

Comment:
IIRC, I was thinking there might be upgrade/downgrade implications involving reading/writing log records for the right version of the log file, but even though I convinced myself that wasn't possible, this way I didn't have even think about it.

I'll add a comment.